src/submission/migration/migration-manager.ts

```diff
@@ -139,6 +139,10 @@ export namespace MigrationManager {
       newSchemaVersion,
```

Comment:
in this line surround this with try catch in the catch you abort the migration. also it would be nice if the migration stored the reason it failed in such case